### PR TITLE
Enabling SPI for STM32F2

### PR DESCRIPTION
**Pull request by ikmdani**
These commits enable SPI on STM32F2. test_spi_loopback was used for testing on SPI-1 and the results is tests passed.

### CODE REVIEW

File: boards/arm/nucleo_f207zg/doc/index.rst [hunk 1]
```diff
@@ -139,6 +141,7 @@ Default Zephyr Peripheral Mapping:
 
 - UART_3 TX/RX : PD8/PD9 (ST-Link Virtual Port Com)
 - UART_6 TX/RX : PG14/PG9 (Arduino Serial)
+- SPI1 NSS/SCK/MISO/MOSI : PA4/PA5/PA6/PA7 (Arduino SPI)
```
Comment (erwango) [inline boards/arm/nucleo_f207zg/doc/index.rst:144]: Please also update `boards/arm/nucleo_f207zg/arduino_r3_connector.dtsi` to configure arduino_spi in dts:
`arduino_spi: &spi1 {};`
When doing so please also update `supported` section in boards yaml to document it supports `arduino_spi`.

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks!
One minor addition required, aside this: LGTM.

**Review by erwango — APPROVED** [review]

**Comment by ikmdani** [comment]
@ABOSTM , i request you to kindly review this GP request.

File: tests/drivers/spi/spi_loopback/boards/nucleo_f207zg.overlay [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2020 STMicroelectronics
```
Comment (FRASTM) [inline tests/drivers/spi/spi_loopback/boards/nucleo_f207zg.overlay:2]: Copyright (c) 202**1** ?

**Review by FRASTM — APPROVED** [review]

**Review by erwango — CHANGES_REQUESTED** [review]
@ikmdani please fix comments in original commits instead of using additional commits to fix comments.

File: tests/drivers/spi/spi_loopback/boards/nucleo_f207zg.overlay (hunk 2 above)
Comment (ikmdani) [reply tests/drivers/spi/spi_loopback/boards/nucleo_f207zg.overlay]: @FRASTM , corrected the year. Thanks.

**Review by erwango — APPROVED** [review]

**Pull request merged**